workflows/cellranger-quant/run-cellranger.nf

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env nextflow
+nextflow.enable.dsl=2
+
+// basic parameters
+params.ref_dir = 's3://nextflow-ccdl-data/reference/homo_sapiens/ensembl-100'
+params.index_dir = 'cellranger_index'
+params.index_name = 'cdna'
+
+params.run_metafile = 's3://ccdl-scpca-data/sample_info/scpca-library-metadata.tsv'
+params.run_ids = "SCPCR000001" //comma separated list to be parsed into a list
+params.outdir = 's3://nextflow-ccdl-results/scpca/cellranger-quant'
+
+// build full paths
+params.index_path = "${params.ref_dir}/${params.index_dir}/${params.index_name}"
+
+process cellranger{
+  container '589864003899.dkr.ecr.us-east-1.amazonaws.com/scpca-cellranger:4.0.0'
+  publishDir "${params.outdir}", mode: 'copy'
+  label 'cpus_8'
+  label 'bigdisk'
+  input:
+    tuple val(id), val(samples), path(fastq_dir)
+    path index
+  output:
+    path output_id
+  script:
+    output_id = "${id}-${index}"
+    """
+    cellranger count \
+      --id=${output_id} \
+      --transcriptome=${index} \
+      --fastqs=${fastq_dir} \
+      --sample=${samples} \
+      --localcores=${task.cpus} \
+      --localmem=${task.memory.toGiga()}
```

Comment:
`task.memory` is the memory requirements that were set by nextflow (in the config file in this case). `toGiga()` is the function that tells it to report the value in gigabytes, because cellranger needs to know how much memory is available expressed in gigabyte units. `task.cpus` is the same, but I don't have to change the units, so it goes in bare.

Reply:
Makes sense!